### PR TITLE
Longest-Substring-Without-Repeating-Characters solution in C#

### PR DESCRIPTION
**Pull request by PatelKeviin**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description

This is the sliding window solution for the 'Longest Substring without Repeating Characters' problem, which is way more efficient than the naive brute force approach. 
Complexity Analysis:
The time complexity for the algorithms is linear i.e. O(n), where n is the length of the given string.
The space complexity is O(k), where k is the number of unique characters in the string.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue: [#563 ](https://github.com/TesseractCoding/NeoAlgo/issues/563)

### CODE REVIEW

**Comment by PatelKeviin** [comment]
Hi @Kajol-Kumari @plazzy99 @SKAUL05 @siddharth25pandey @iamrajiv, Could any 3 of you give valuable feedback on my PR. Thanks in advance!

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@PatelKeviin add sample input-output, space and time complexity

File: C-Sharp/cp/LengthOfLongestSubstring.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
+﻿public class Solution
```
Comment (atarax665) [inline C-Sharp/cp/LengthOfLongestSubstring.cs]: Add description of the problem.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: C-Sharp/cp/LengthOfLongestSubstring.cs (hunk 1 above)
Comment (PatelKeviin) [reply C-Sharp/cp/LengthOfLongestSubstring.cs]: Hi @atarax665, I hope this description works. Thanks for your time and effort.

**Comment by PatelKeviin** [comment]
> @PatelKeviin add sample input-output, space and time complexity

Hi @iamrajiv, I have added sample inputs/outputs as well as complexities for the algorithm. I hope it answers your concern. Thanks!

**Review by iamrajiv — APPROVED** [review]

**Review by SKAUL05 — APPROVED** [review]

File: C-Sharp/cp/LengthOfLongestSubstring.cs (hunk 1 above)
Comment (PatelKeviin) [reply C-Sharp/cp/LengthOfLongestSubstring.cs]: Hi @atarax665, could you please review my contribution. Thanks in advance!

**Comment by PatelKeviin** [comment]
Hi @atarax665, hope this message finds you well. Could you please provide your valuable feedback on this issue so that I can update/improve it and close it asap. Thank you :)

**Comment by PatelKeviin** [comment]
Hi @iamrajiv and @SKAUL05 thank you for approving my changes. I am still waiting on @atarax665 for quite a few days now (more than 10) and am wondering if I could get some feedback and get this merged. Could you please do something about it. Thanks once again :)

**Review by atarax665 — APPROVED** [review]
Code looks good to me, sorry for the delay though.

**Comment by atarax665** [comment]
@all-contributors add @PatelKeviin  for his code.

**Pull request merged**